include/picongpu/fields/incidentField/Solver.kernel

```diff
@@ -90,8 +136,128 @@ namespace picongpu
 
                     //! Current time step, in iterations; can be fractional
                     float_X currentStep;
-                };
 
+                    /** Direction of the incident field propagation
+                     *
+                     * +1._X is positive direction (from the min boundary inwards).
+                     * -1._X is negative direction (from the max boundary inwards)
+                     */
+                    float_X direction;
+
+                    //! Boundary axis, 0 = x, 1 = y, 2 = z
+                    static constexpr uint32_t axis = T_axis;
+
+                    //! Whether the updated field is total or scattered
+                    bool isUpdatedFieldTotal;
+
+                private:
+                    /** Calculate correction for the updated field due to incident field for the given shift
+                     *

```

Comment:
(2a) Please explain in more detail how the selection for the 'nodes on the opposite site of the Huygens surface' works.

(2b) Name the coefficients that are derived from the curl coefficients.

(2c) Name the total coefficient in front of incident field.

(2d) Describe the parameters used to select the correct component of the incident fields.

(2e) Say that coeff1 and coeff2 are vectors which are multiplied with the respective incidentField components in order to calculate the updatedField correction as a vector.
You can also explain the reasoning, i.e. that this allows to generalize the calculation to the calculation of vector triple sum instead of multiple implementations of triple sums for every updated field component and axes

Reply:
(3) With schemes you mean the different types of Maxwell solver? If so, this is what should be written.

Reply:
(4) Please correct `betweeen` (3 'e').

Reply:
Sorry, that is quite some text to add. But this is what I feel is necessary in addition to the readthedocs documentation to understand the implementation.

Reply:
Done in a combination of this function and a new description of the vector update when these members are declared.